clusterdock/cluster.py

```diff
@@ -148,6 +148,13 @@ def __iter__(self):
     def __len__(self):
         return len(self.nodes)
 
+    def get_node_groups_by_names(self, node_group_names):
```

Comment:
As we discussed, I don't think either of these functions are necessary.

1) NodeGroups should be uniquely named, so not sure there's a need for a function that does this.

2) NodeGroup itself has an `ssh` method (and its nodes are accessible via the `nodes` attribute), so I don't think we need `get_nodes_by_node_group_names`, either.